### PR TITLE
HexEditor+Playground+Spreadsheet: Use `GUI::MessageBox::ask_about_unsaved_changes()`

### PR DESCRIPTION
**Pull request by krkk**

### CODE REVIEW

**Comment by krkk** [comment]
Changes since last push:
- tried to reword commits as of @trflynn89’s request in #12544, but I don't think it's much better. sorry.
- rebased to master

**Comment by trflynn89** [comment]
Sorry, after merging #12544, this one now has a conflict

File: Userland/Applications/HexEditor/HexEditorWidget.cpp [hunk 1]
```diff
@@ -344,14 +339,14 @@ bool HexEditorWidget::request_close()
...
-        return false;
+    auto result = GUI::MessageBox::ask_about_unsaved_changes(window(), m_path);
     if (result == GUI::MessageBox::ExecYes) {
         m_save_action->activate();
         return m_document_dirty == false;
     }
-    return true;
+    if (result == GUI::MessageBox::ExecNo)
+        return true;
+    return false;
```
Comment (trflynn89) [inline Userland/Applications/HexEditor/HexEditorWidget.cpp:349]: Why not:
```c++
return result == GUI::MessageBox::ExecNo;
```

File: Userland/Applications/Spreadsheet/SpreadsheetWidget.cpp [hunk 2]
```diff
@@ -421,8 +421,7 @@ bool SpreadsheetWidget::request_close()
     if (!m_workbook->dirty())
         return true;
 
-    auto result = GUI::MessageBox::show(window(), "The spreadsheet has been modified. Would you like to save?", "Unsaved changes", GUI::MessageBox::Type::Warning, GUI::MessageBox::InputType::YesNoCancel);
-
+    auto result = GUI::MessageBox::ask_about_unsaved_changes(window(), current_filename());
```
Comment (trflynn89) [inline Userland/Applications/Spreadsheet/SpreadsheetWidget.cpp:424]: This commit isn't really "unifying" any popups (at least, when I read the commit title, I thought the commit would be unifying some duplicated behavior within the app itself).

I think better language would be "Use the system-wide unsaved changes dialog".

**Comment by krkk** [comment]
Changes since last push:
- rebased and resolved conflicts
- applied suggested changes

**Pull request merged**